### DISCUSSION TITLE
Event button locks sometimes

### DISCUSSION THREAD

**Discussion by grabowc** [artisan-roaster-scope/artisan#1201] [Q&A] [answered]
Occasionally I will hit Event and then when I try to choose an Event like Air or Burner, it won't register - it just locks until I hit ESC.

Any others seeing this? In Artisan 2.8.2 for MacOS.

Thanks

**Comment by MAKOMO** [comment]
I assume you are using the standard Event button (menu Config >> Events, 1st tab "Event Button") and the Events Editor line to specify the event type via its popup. Right?

When exactly does it lock. On clicking that Event Type popup? Is the popup shown? Or is the lockup happens on pressing "Update" in the Events Editor?

<img width="798" alt="Screenshot 2023-06-13 at 16 40 51" src="https://github.com/artisan-roaster-scope/artisan/assets/13909011/7673cf7e-c872-4bec-a5ac-3dc3fa276c69">

**Comment by grabowc** [comment]
Yep, just like that. It locks on the Event Type popup (it is shown) — so I will choose Air from the popup for example but it just sits there until I hit ESC.

**Comment by MAKOMO** [comment, marked as answer]
If it hangs on the popup (shown) this is most likely an issue in the underlying GUI toolkit we are using. This was recently updated. Do you experience the same issue using the Release Candidate for the next Artisan version?

https://github.com/artisan-roaster-scope/artisan/releases/tag/v2.8.3-RC

**Reply by grabowc** [reply]
Subsequent updates fixed this. Thanks!
